### PR TITLE
[IS34] Add bottom navigation bar

### PR DESCRIPTION
**Pull request by Wos2610**
# *SUMARY*
Add new bottom navigation bar.

# *CHANGE*
Change in some files:
- main.dart: 
- [x] Change MyApp() to CloneInstagramApp().
- main_app.dart:
- [x] Create CloneInstagramApp.
- pages/dashboard_page.dart
- [x] Create DashboardPage class and _DashboardPageState class.
- [x] Create BottomNavigationBar.
- constants/SourceString.dart
- [x] Create SourceString class.

# *CHECKLIST*
- [x] Add bottom navigation bar.
- [x] Add SourceString file.
- [x] Split main.dart to some files.

# *RESULT*
image

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -1,18 +1,110 @@
...
       title: 'Clone Instagram',
+      theme: ThemeData(
+        useMaterial3: true,
+      ),
+      home: const DashboardPage(),
     );
   }
 }
 
+class DashboardPage extends StatefulWidget {
```
Comment (Tris1702) [inline lib/main.dart]: Should write in a separate file pages/dashboard_page.dart

File: lib/main.dart [hunk 2]
```diff
@@ -1,18 +1,110 @@
 import 'package:flutter/material.dart';
 
 void main() {
-  runApp(const MyApp());
+  runApp(const CloneInstagramApp());
 }
 
-class MyApp extends StatelessWidget {
-  const MyApp({super.key});
+class CloneInstagramApp extends StatelessWidget {
```
Comment (Tris1702) [inline lib/main.dart]: Should write in a separate file such as: main_app.dart

File: lib/main.dart [hunk 3]
```diff
@@ -1,18 +1,110 @@
...
+}
+
+class _DashboardPageState extends State<DashboardPage> {
+  int _selectedIndex = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          title: const Text('Clone Instagram'),
```
Comment (Tris1702) [inline lib/main.dart]: All strings should be defined in a string source file. In the most simple way is define a class `SourceString` with `static const appName = "Clone Instagram"`
When you want to use it, just call `SourceString.appName`

**Review by Tris1702 — CHANGES_REQUESTED** [review]
As I see, all colors are not correct. Please fix it

File: lib/main.dart (hunk 1 above)
Comment (Wos2610) [reply lib/main.dart]: I changed

File: lib/main.dart (hunk 2 above)
Comment (Wos2610) [reply lib/main.dart]: I changed

File: lib/main.dart (hunk 3 above)
Comment (Wos2610) [reply lib/main.dart]: I changed

**Review by mdunggggg — APPROVED** [review]

File: lib/pages/dashboard_page.dart [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  int _selectedIndex = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          title: const Text(SourceString.titleAppBar),
+        ),
+        body: const Center(
+          child: Text(""),
```
Comment (Tris1702) [inline lib/pages/dashboard_page.dart]: Body should be the Screen that was selected

File: lib/pages/dashboard_page.dart [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+              label: SourceString.profile,
+            ),
+          ],
+          currentIndex: _selectedIndex,
+          selectedItemColor: Colors.black,
+          unselectedItemColor: Colors.black,
+          onTap: (int index){
+            switch(index){
+              case 0:
+              // Navigator.push(
```
Comment (Tris1702) [inline lib/pages/dashboard_page.dart]: when changing index, only the body changes, not pushing a screen on top. This behavior isn't right

**Review by Tris1702 — CHANGES_REQUESTED** [review]

File: lib/pages/dashboard_page.dart (hunk 4 above)
Comment (Wos2610) [reply lib/pages/dashboard_page.dart]: I changed

File: lib/pages/dashboard_page.dart (hunk 5 above)
Comment (Wos2610) [reply lib/pages/dashboard_page.dart]: I changed

**Pull request closed without merging**